src/Prescription/Prescription.service.ts

```diff
@@ -0,0 +1,135 @@
+import {
+  Injectable,
+  NotFoundException,
+  InternalServerErrorException,
+} from '@nestjs/common';
+import { InjectModel } from '@nestjs/mongoose';
+import { Model } from 'mongoose';
+import { GetPrescriptionDto } from './dto/getPrescription.dto';
+import { IPrescription } from './Interfaces/prescription.interfaces';
+import { CreatePrescriptionDto } from './dto/createPrescription.dto';
+import { Prescription } from 'src/Schema/Prescription.schema';
+@Injectable()
+export class PrescriptionService {
+  constructor(
+    @InjectModel('PrescriptionSchema')
+    private prescriptionModel: Model<IPrescription>, // Injection of the Prescription model,
+  ) {}
+
+  /**
+   * Fetches prescriptions for a specific patient from the database.
+   * The number of prescriptions returned and the page number are defined in the payload.
+   *
+   * @param {GetPrescriptionDto} payload - The query parameters for fetching the prescriptions.
+   * @throws {NotFoundException} - Throws an exception if no prescriptions are found for the given NHI.
+   * @returns {Promise<Array<IPrescription>>} - A promise that resolves to an array of prescriptions.
+   */
+  async searchPrescriptionsForPatient(
+    payload: GetPrescriptionDto,
+  ): Promise<Array<IPrescription>> {
+    try {
+      const { nhi, size = 10, page = 1 } = payload; // Default values for size and page if not provided
+
```

Comment:
Let's add one more condition in here, where we will be checking for the max allowed size ( e.g. 50 ) and in case size become greater than that size then the max allowed size will get set